src/viem/prepareTransactions.test.ts

```diff
@@ -0,0 +1,332 @@
+import BigNumber from 'bignumber.js'
+import { Network, NetworkId } from 'src/transactions/types'
+import { TokenBalanceWithAddress } from 'src/tokens/slice'
+import {
+  getFeeCurrencyAddress,
+  getMaxGasCost,
+  prepareERC20TransferTransaction,
+  prepareTransactions,
+  tryEstimateTransaction,
+  tryEstimateTransactions,
+} from 'src/viem/prepareTransactions'
+import { Address, BaseError, encodeFunctionData, EstimateGasExecutionError } from 'viem'
+import { estimateFeesPerGas } from 'src/viem/estimateFeesPerGas'
+import { publicClient } from 'src/viem/index'
+import mocked = jest.mocked
+import { TransactionRequestCIP42 } from 'node_modules/viem/_types/chains/celo/types'
+import erc20 from 'src/abis/IERC20'
+
+jest.mock('src/viem/estimateFeesPerGas')
+jest.mock('viem')
+
+describe('prepareTransactions module', () => {
+  const mockFeeCurrencies: TokenBalanceWithAddress[] = [
+    {
+      address: '0xfee1',
+      balance: new BigNumber(100), // 10k wei, 100.0 decimals
```

Comment:
nit: should we use `units` instead? `wei` is really 10^-18 ETH.
Otherwise the step to name functions or variables with `wei` as a suffix / prefix isn't far. And may really confuse us.

Or has `wei` been really used interchangeably to talk about the smallest unit of any token?

Reply:
that's a good point, I made the same mistake in `prepareERC20TransferTransaction`. wei is the smallest unit of eth, and I've seen it used as the smallest units of other tokens that use 18 decimal places, but I don't think it applies across the board for tokens that use different numbers of decimal places.

Reply:
I renamed and added docs to `prepareERC20TransferTransaction`